src/pepr/docs-gen/main.ts

```diff
@@ -0,0 +1,207 @@
+// Copyright 2024 Defense Unicorns
+// SPDX-License-Identifier: AGPL-3.0-or-later OR LicenseRef-Defense-Unicorns-Commercial
+import * as fs from "fs/promises";
+import * as path from "path";
+
+// Constants for styling and configuration
+const INDENT_SIZE = 20;
+const MAX_HEADER_LEVEL = 6;
+const MAX_DEPTH = 10;
+const OUTPUT_DIR = "./docs/reference/configuration/custom-resources/";
+const TABLE_STYLE = 'style="width: 100%; table-layout: fixed;"';
+
+// Utility to capitalize the first letter of a string
+const capitalizeFirstLetter = (text: string): string =>
+  text.charAt(0).toUpperCase() + text.slice(1);
+
+// Helper to format a field row (avoiding extra spacing)
+const formatRow = (field: string, type: string, description: string): string =>
+  `<tr><td style="white-space: nowrap;">${field}</td><td style="white-space: nowrap;">${type}</td><td>${description}</td></tr>`;
+
+// Helper function to generate a header level (capped at MAX_HEADER_LEVEL)
+const getHeaderLevel = (depth: number): string => "#".repeat(Math.min(depth, MAX_HEADER_LEVEL));
+
+/**
+ * Generates a Markdown table with sections and indentation (non-collapsible)
+ */
+function generateTable(title: string, rows: string[], currentDepth: number): string {
+  const capitalizedTitle = capitalizeFirstLetter(title);
+  const headerLevel = getHeaderLevel(currentDepth);
+  const indentStyle = `style="margin-left: ${currentDepth * INDENT_SIZE}px; padding-top: 30px;"`;
+
+  return `
+<a id="${capitalizedTitle}"></a>
+<div ${indentStyle}>
+
+${headerLevel} ${capitalizedTitle}
+<table ${TABLE_STYLE}>
+  <thead>
+    <tr>
+      <th style="width: 20%; white-space: nowrap;">Field</th>
+      <th style="width: 25%; white-space: nowrap;">Type</th>
+      <th style="width: 55%; white-space: nowrap;">Description</th>
+    </tr>
+  </thead>
+  <tbody>
+    ${rows.join("")}
+  </tbody>
+</table>
+</div>
+`;
+}
+
+// Helper to handle arrays
+function handleArray(
+  schema: SchemaProperty,
+  field: string,
+  currentDepth: number,
+): { type: string; markdown: string } {
+  let type = "";
+  let markdown = "";
+
+  if (schema.items?.enum) {
+    type = `Policy[] (enum):<ul>${schema.items.enum.map((value: string) => `<li><code>${value}</code></li>`).join("")}</ul>`;
+  } else if (schema.items?.properties) {
+    const capitalizedField = capitalizeFirstLetter(field);
```

Comment:
```suggestion
  const capitalizedField = capitalizeFirstLetter(field);
  if (schema.items?.enum) {
    type = `${capitalizedField}[] (enum):<ul>${schema.items.enum.map((value: string) => `<li><code>${value}</code></li>`).join("")}</ul>`;
  } else if (schema.items?.properties) {
```

Might suggest this to future proof the logic here, it will update from `Policy` to `Policies` but otherwise does the same thing.